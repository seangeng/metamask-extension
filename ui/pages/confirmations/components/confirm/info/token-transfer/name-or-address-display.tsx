import { NameType } from '@metamask/name-controller';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useFallbackDisplayName } from '../../../../../../components/app/confirm/info/row/hook';
import NicknamePopovers from '../../../../../../components/app/modals/nickname-popovers';
import Name from '../../../../../../components/app/name';
import {
  AvatarAccount,
  AvatarAccountSize,
  Box,
  Text,
} from '../../../../../../components/component-library';
import {
  AlignItems,
  BorderColor,
  Display,
  FlexDirection,
  TextColor,
} from '../../../../../../helpers/constants/design-system';
import { getPetnamesEnabled } from '../../../../../../selectors';

export const NameOrAddressDisplay = ({ address }: { address: string }) => {
  const isPetNamesEnabled = useSelector(getPetnamesEnabled);
  const { displayName, hexAddress } = useFallbackDisplayName(address);
  const [isNicknamePopoverShown, setIsNicknamePopoverShown] = useState(false);
  const handleDisplayNameClick = () => setIsNicknamePopoverShown(true);
  const onCloseHandler = () => setIsNicknamePopoverShown(false);

  if (isPetNamesEnabled) {
    return <Name value={hexAddress} type={NameType.ETHEREUM_ADDRESS} />;
  }

  return (
    <>
      <Box
        display={Display.Flex}
        flexDirection={FlexDirection.Row}
        alignItems={AlignItems.center}
        onClick={handleDisplayNameClick}
      >
        <AvatarAccount
          address={address}
          size={AvatarAccountSize.Xs}
          borderColor={BorderColor.transparent}
        />
        <Text
          marginLeft={2}
          color={TextColor.inherit}
          data-testid="confirm-info-row-display-name"
        >
          {displayName}
        </Text>
      </Box>
      {isNicknamePopoverShown ? (
        <NicknamePopovers onClose={onCloseHandler} address={hexAddress} />
      ) : null}
    </>
  );
};
