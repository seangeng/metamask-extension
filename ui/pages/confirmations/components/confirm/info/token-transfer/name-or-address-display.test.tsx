import React from 'react';
import configureMockStore from 'redux-mock-store';
import { getMockTokenTransferConfirmState } from '../../../../../../../test/data/confirmations/helper';
import { renderWithConfirmContextProvider } from '../../../../../../../test/lib/confirmations/render-helpers';
import { NameOrAddressDisplay } from './name-or-address-display';

jest.mock(
  '../../../../../../components/app/alert-system/contexts/alertMetricsContext',
  () => ({
    useAlertMetrics: jest.fn(() => ({
      trackAlertMetrics: jest.fn(),
    })),
  }),
);

describe('NameOrAddressDisplay', () => {
  it('renders correctly', () => {
    const state = getMockTokenTransferConfirmState({});
    const mockStore = configureMockStore([])(state);
    const { container } = renderWithConfirmContextProvider(
      <NameOrAddressDisplay address="0x388C818CA8B9251b393131C08a736A67ccB19297" />,
      mockStore,
    );
    expect(container).toMatchSnapshot();
  });
});
