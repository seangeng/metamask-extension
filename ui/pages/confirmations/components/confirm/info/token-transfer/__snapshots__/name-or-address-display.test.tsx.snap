// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameOrAddressDisplay renders correctly 1`] = `
<div>
  <div
    class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
  >
    <div
      class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
    >
      <div
        class="mm-avatar-account__jazzicon"
      >
        <div
          style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(1, 121, 142);"
        >
          <svg
            height="16"
            width="16"
            x="0"
            y="0"
          >
            <rect
              fill="#F5E000"
              height="16"
              transform="translate(-1.1691021707883256 -4.615852707363339) rotate(318.6 8 8)"
              width="16"
              x="0"
              y="0"
            />
            <rect
              fill="#1881F2"
              height="16"
              transform="translate(0.31329425325262034 -5.330574883433899) rotate(399.1 8 8)"
              width="16"
              x="0"
              y="0"
            />
            <rect
              fill="#234FE1"
              height="16"
              transform="translate(2.705792259754136 -12.770998732484898) rotate(402.4 8 8)"
              width="16"
              x="0"
              y="0"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
      data-testid="confirm-info-row-display-name"
    >
      0x388C8...19297
    </p>
  </div>
</div>
`;
